File: ios-badania-app/BadaniaApp/BadaniaApp/Views/SumView.swift
//
//  SumView.swift
//  BadaniaApp
//
//  Panel z sumą wybranych badań
//

import SwiftUI

struct SumView: View {
    let sum: Double
    let formattedSum: String
    
    var body: some View {
        VStack(spacing: 0) {
            Divider()
            
            HStack {
                Text("Suma wybranych badań:")
                    .font(.headline)
                    .foregroundColor(Color(hex: "212121"))
                
                Spacer()
                
                Text(formattedSum)
                    .font(.system(size: 24, weight: .bold))
                    .foregroundColor(Color(hex: "1976d2"))
            }
            .padding(.horizontal, 20)
            .padding(.vertical, 16)
            .background(Color(.systemBackground))
        }
    }
}

File: android-badania-app/app/src/main/java/com/decentcode/badaniaapp/ui/components/SumView.kt
package com.decentcode.badaniaapp.ui.components

import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.decentcode.badaniaapp.utils.Color

/**
 * Komponent wyświetlający sumę wybranych badań
 */
@Composable
fun SumView(
    sum: Double,
    formattedSum: String,
    modifier: Modifier = Modifier
) {
    Surface(
        modifier = modifier.fillMaxWidth(),
        color = MaterialTheme.colorScheme.surface,
        shadowElevation = 4.dp
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 20.dp, vertical = 16.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text(
                text = "Suma wybranych badań:",
                style = MaterialTheme.typography.titleMedium,
                fontWeight = FontWeight.Medium,
                color = Color("212121")
            )
            
            Text(
                text = formattedSum,
                style = MaterialTheme.typography.headlineSmall,
                fontWeight = FontWeight.Bold,
                color = Color("1976d2")
            )
        }
    }
}

